src/Umbraco.Web/PublishedCache/NuCache/DataSource/NuCacheCompressionOptions.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Umbraco.Web.PublishedCache.NuCache.DataSource
+{
+    public struct NuCacheCompressionOptions : IEquatable<NuCacheCompressionOptions>
```

Comment:
This could also be an interface that takes the document / media and lets the implementer determine per document whether to compress or not. E.x. compress documents older than 90 days.

Reply:
Is there any reason you would want to compress property values based on time based logic like that? Seems to me like having all complex editors that store JSON or rich text should just be compressed - either on or off. I guess I'm just wondering what the benefit of not compressing 'new' rich text would be?

Reply:
Saves on decompressing frequently used content when requests arrive. LZ4 seems fast enough though that this may not be an issue, but I can't say for sure for everyone.

Reply:
Potentially you could even remove properties and retrieve them from elsewhere.

Reply:
> Saves on decompressing frequently used content when requests arrive. LZ4 seems fast enough though that this may not be an issue, but I can't say for sure for everyone.

Cool I just wanted to check if I was missing something. I'll need to do some rendering benchmarks with the decompression happening for properties and see what kind of difference that makes.

Reply:
Potentially it might be better to just copy the compressed messagepack bytes straight to bplustree rather than deserializing messagepack, serialising again using messagepack and or the primitive serializers. It would need benchmarking, but given the data would still be compressed it may be faster to read off the disk doing it this way. Alternatively, if [ZeroFormatter](https://github.com/neuecc/ZeroFormatter) is suitable (not sure, never used it), it supports lazy deserialization at the cost of a larger size, could be used as well.